src/algebraic_geometry/structure_sheaf.lean

```diff
@@ -799,6 +799,32 @@ def basic_open_iso (f : R) : (structure_sheaf R).presheaf.obj (op (basic_open f)
   CommRing.of (localization.away f) :=
 (as_iso (show CommRing.of _ ⟶ _, from to_basic_open R f)).symm
 
+@[elementwise] lemma to_global_factors : to_open R ⊤ =
+  @category_struct.comp CommRing _ (CommRing.of R) (CommRing.of _) (CommRing.of _)
+    (show CommRing.of _ ⟶ CommRing.of _, from (algebra_map R (localization.away (1 : R))))
+  (to_basic_open R (1 : R)) ≫  (structure_sheaf R).presheaf.map (eq_to_hom (by simp)).op :=
```

Comment:
If you replace `simp` with `squeeze_simp`, you can see that the simp is just applying the lemma `basic_open_one`. Can you replace the `by simp` with this lemma?